### PR TITLE
Driver/gentec maestro

### PR DESCRIPTION
**Pull request by SvenBo90**
feat: New driver

Driver for Gentec Maestro and example notebook

no breaking changes

### CODE REVIEW

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        self.add_parameter('wavelength',
+                           get_cmd='*GWL',
+                           set_cmd='*PWC{0:0>5}',
+                           get_parser=wavelength_get_parser,
+                           label='wavelength',
+                           unit='nm')
+
+        def zero_offset_get_parser(ans):
+            return int(ans[5:])
+        self.add_parameter('zero_offset_status',
```
Comment (astafan8) [inline qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: would `zero_offset_enabled` be a better name?

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                           get_cmd='*GZO',
+                           get_parser=zero_offset_get_parser,
+                           val_mapping=create_on_off_val_mapping(on_val=1, off_val=0),
+                           label='zero offset status')
```
Comment (astafan8) [inline qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: please capitalize the parameter labels

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        # add functions
+        self.add_function('set_zero_offset',
```
Comment (astafan8) [inline qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: Please create bound methods for these two, [we are in the process of deprecating functions](http://qcodes.github.io/Qcodes/examples/Creating%20Instrument%20Drivers.html#Functions).

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        self.add_function('clear_zero_offset',
+                          call_cmd='*COU')
+
+        # print connect message
+        self.connect_message(idn_param='IDN')
+
+    # get methods
+    def get_idn(self):
+        return {'vendor': 'Gentec', 'model': self.model,
+                'firmware': str(self.firmware_version)}
```
Comment (astafan8) [inline qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: why making `self.firmware_version` a string here, and not in `__init__`? is the numeric value useful anywhere?

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    def __init__(self, name, address, baud_rate=115200, **kwargs):
+
+        super().__init__(name, address, **kwargs)
+
+        # set baud rate
+        self.visa_handle.baud_rate = baud_rate
+
+        # get instrument information
+        self.model = self.visa_handle.query('*VER').split()[0]
+        self.firmware_version = self.visa_handle.query('*VER').split()[2]
```
Comment (astafan8) [inline qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: Please make a spearate (private?) method for this query `self.visa_handle.query('*VER')` and re-use it here. (perhaps, smth like `model, _, fw_version = self._query_versions()`)

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py (hunk 1 above)
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: I did change the name of the parameter.

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py (hunk 2 above)
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: I did capitalize all parameter labels.

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py (hunk 3 above)
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: They are now bound methods.

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py (hunk 4 above)
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: It's actually a string already so no need to convert it.

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        # print connect message
+        self.connect_message(idn_param='IDN')
+
+    # get methods
+    def get_idn(self):
+        return {'vendor': 'Gentec', 'model': self.model, 'firmware': self.firmware_version}
+
+    # further methods
+    def clear_zero_offset(self):
+        self.visa_handle.write('*COU')
```
[the remark this answers is not in the document]
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: Done!

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    def set_zero_offset(self):
+        self.visa_handle.write('*SOU')
+
+    def _query_versions(self):
+        query_return = self.visa_handle.query('*VER').split()
+        return query_return[0], query_return[1], query_return[2]
```
[the remark this answers is not in the document]
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: Done!

File: qcodes/instrument_drivers/Gentec/Gentec_Maestro.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        def zero_offset_get_parser(ans):
+            return int(ans[5:])
+        self.add_parameter('zero_offset_enabled',
+                           get_cmd='*GZO',
+                           get_parser=zero_offset_get_parser,
+                           val_mapping=create_on_off_val_mapping(on_val=1, off_val=0),
+                           label='Zero offset enabled')
+
+        # print connect message
+        self.connect_message(idn_param='IDN')
```
[the remark this answers is not in the document]
Comment (SvenBo90) [reply qcodes/instrument_drivers/Gentec/Gentec_Maestro.py]: Done!

**Pull request merged**